src/library/playlistfeature.cpp

```diff
@@ -189,15 +189,31 @@ void PlaylistFeature::slotPlaylistTableChanged(int playlistId) {
     enum PlaylistDAO::HiddenType type = m_playlistDao.getHiddenType(playlistId);
     if (type == PlaylistDAO::PLHT_NOT_HIDDEN ||
         type == PlaylistDAO::PLHT_UNKNOWN) { // In case of a deleted Playlist
+        // TODO: Is this even necessary?
         clearChildModel();
         m_lastRightClickedIndex = constructChildModel(playlistId);
+    }
+}
+
+void PlaylistFeature::slotPlaylistTableRenamed(int playlistId,
```

Comment:
they're doing identical logic in either case -- the goal is 1) refresh the sidebar 2) activate the playlist so it can be the same in both